security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -335,6 +356,51 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	return ns, nil
 }
 
+func (sc *SecretCache) addFileWatcher(file string, key ConnKey) {
+	// Check if this file is being already watched, if so ignore it. FileWatcher has the functionality of
+	// checking for duplicates. This check is needed here to avoid processing duplicate events for the same file.
+	sc.certMutex.Lock()
+	_, exists := sc.fileCerts[file]
+	if !exists {
+		sc.fileCerts[file] = struct{}{}
+	}
+	sc.certMutex.Unlock()
+	// File is not being watched, start watching now and trigger key push.
+	if !exists {
+		cacheLog.Debugf("adding watcher for file %s", file)
+		if err := sc.certWatcher.Add(file); err != nil {
+			cacheLog.Errorf("error adding watcher for file %s, Skipping watches ", file)
+			return
+		}
+		go func() {
+			var timerC <-chan time.Time
+			for {
+				select {
+				case <-timerC:
+					timerC = nil
+					if val, ok := sc.secrets.Load(key); ok {
+						// Trigger the callback that pushes the secrets to proxy.
+						secret := val.(model.SecretItem)
+						cacheLog.Debugf("file changed %s, triggering push to proxy", file)
+						sc.callbackWithTimeout(key, &secret)
+					}
+				case e := <-sc.certWatcher.Events(file):
+					if e.Op&fsnotify.Remove == fsnotify.Remove {
+						// File is deleted. Stop watching
+						cacheLog.Debugf("file deleted %s, removing the watcher", file)
```

Comment:
if its later added back we should re add a watch. I believe this does occur as expected, just want to verify

Reply:
This is a good point. I removed the deletion logic because some systems delete and readd the new file. so it is safe not to remove from watch list. I also added a TODO to properly cleanup stale watches later.